pkg/auto/validate.go

```diff
@@ -170,3 +179,45 @@ func (v *validation) checkNeverModifyReleasedChart(assetFilePaths map[string]str
 
 	return nil
 }
+
+// CompareIndexFiles will load the current index.yaml file from the root filesystem and compare it with the index.yaml file from charts.rancher.io
+func CompareIndexFiles(rootFs billy.Filesystem) error {
```

Comment:
This function returns an error in some cases and also exits in other cases. There should only be 1 entry and exit point for a program, generally. Can we update this to return an error consistently and have the caller made the decision to exit or otherwise?

Reply:
Fixed